api/server.ts

```
@@ -2,21 +2,56 @@ import { z } from "zod";
+import { handleGetArtistProfile } from "../lib/handlers/artistProfileHandler";
 import { TOOL_CONFIGS } from "../lib/toolConfigs";
 
 const handler = initializeMcpApiHandler(
   (server) => {
+    // Add global error handler
+    server.server.onerror = (error) => {
+      console.error("[MCP Server] Error:", error);
+    };
+
+    // Wrap tool handlers with error logging
+    const wrapHandler = (handler: Function, toolName: string) => {
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)